java-checks-test-sources/default/src/main/java/checks/spring/CacheAnnotationsShouldOnlyBeAppliedToConcreteClassesCheckSample.java

```diff
@@ -0,0 +1,89 @@
+package checks.spring;
+
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.CachePut;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.annotation.Caching;
+
+public class CacheAnnotationsShouldOnlyBeAppliedToConcreteClassesCheckSample {
+
+  @Cacheable("aCache") // Noncompliant
+  public interface InterfaceCacheable {
+    @Cacheable("aCache") // Noncompliant {{Move this "@Cacheable" annotation from interface to a concrete class.}}
+  //^^^^^^^^^^^^^^^^^^^^
+    String getData(String id);
+  }
+
+
+  @Cacheable("aCache")
+  abstract class AbstractClassCacheable {
+    @Cacheable("aCache")
```

Comment:
After some testing I can confirm that 2 different implementations of an abstract class will have conflicts with the cache if it's configured this way, so we should raise also in this case. Let's discuss the specifics and I can share my experiment